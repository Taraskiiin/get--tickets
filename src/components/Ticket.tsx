import React, { Fragment } from 'react';

import {
  Card,
  Header,
  Price,
  ColumnInfo,
  ColumnInfoTitle,
  ColumnInfoTime,
  Content,
} from '../styles/Ticket';

export interface ITicket {
  ticket: {
    price: number;
    logo: string;
    rows: {
      from: string;
      to: string;
      arrivesAt: string;
      departsAt: string;
    }[];
  };
}

const Ticket: React.FC<ITicket> = ({ ticket }) => {
  return (
    <Card>
      <Header>
        <Price>{ticket.price}&nbsp;P</Price>
        <img src={ticket.logo} alt="logo" />
      </Header>
      <Content>
        {ticket.rows.map((row, index) => (
          <Fragment key={index}>
            <ColumnInfo>
              <ColumnInfoTitle>
                {row.from} - {row.to}
              </ColumnInfoTitle>
              <span>
                {row.departsAt} - {row.arrivesAt}
              </span>
            </ColumnInfo>
            <ColumnInfo>
              <ColumnInfoTitle>В Пути</ColumnInfoTitle>
              <ColumnInfoTime>
                {/**TODO: add function thst calculates time of the trip */}21ч
                15м
              </ColumnInfoTime>
            </ColumnInfo>
            <ColumnInfo>
              <ColumnInfoTitle>
                Пересадки{/**TODO: add count of transplantation */}{' '}
              </ColumnInfoTitle>
              <span>hkg, jnb</span>
            </ColumnInfo>
          </Fragment>
        ))}
      </Content>
    </Card>
  );
};

export default Ticket;
