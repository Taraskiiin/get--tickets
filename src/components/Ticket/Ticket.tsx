import React from "react";
import styled from "styled-components";

export interface ITicket {
  ticket: {
    price: number;
    logo: string;
    rows: {
      from: string;
      to: string;
      arrivesAt: string;
      departsAt: string;
    }[];
  };
}

const Ticket: React.FC<ITicket> = ({ ticket }) => {
  return (
    <Card>
      <Header>
        <Price>{ticket.price}&nbsp;P</Price>
        <img src={ticket.logo} alt="logo" />
      </Header>
      <Content>
        {ticket.rows.map((row) => (
          <>
            <ColumnInfo>
              <ColumnInfoTitle>
                {row.from} - {row.to}
              </ColumnInfoTitle>
              <span>
                {row.departsAt} - {row.arrivesAt}
              </span>
            </ColumnInfo>
            <ColumnInfo>
              <ColumnInfoTitle>В Пути</ColumnInfoTitle>
              <ColumnInfoTime>
                {/**TODO: add function thst calculates time of the trip */}21ч
                15м
              </ColumnInfoTime>
            </ColumnInfo>
            <ColumnInfo>
              <ColumnInfoTitle>
                Пересадки{/**TODO: add count of transplantation */}{" "}
              </ColumnInfoTitle>
              <span>hkg, jnb</span>
            </ColumnInfo>
          </>
        ))}
      </Content>
    </Card>
  );
};

const Card = styled.div`
  background-color: #fff;
  height: max-content;
  border-radius: 10px;
  padding: 25px 50px 25px 20px;
  text-transform: uppercase;
  font-size: 12px;
  margin-bottom: 20px;
`;

const Header = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  align-items: center;
  font-size: 12px;
  margin-bottom: 10px;
`;

const Content = styled.section`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
`;

const ColumnInfo = styled.p`
  > span {
    display: block;
  }
`;

const ColumnInfoTitle = styled.span`
  color: #a0b0b9;
`;

const ColumnInfoTime = styled.span`
  text-transform: lowercase;
`;

const Price = styled.span`
  font-size: 18px;
  color: #2196f3;
`;
export default Ticket;
