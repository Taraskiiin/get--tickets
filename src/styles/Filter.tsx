import styled from 'styled-components';
import { ButtonGroup, Button } from 'react-bootstrap';

export const Nav = styled(ButtonGroup)`
  width: 100%;
  height: max-content;
  display: flex;
  justify-content: space-between;
  text-transform: uppercase;
  letter-spacing: 0.5px;
  border-radius: 10px;
  background-color: #fff;
  margin-bottom: 20px;
`;

export const StyledButton = styled(Button)`
  text-decoration: none;
  text-align: center;
  padding: 14px 0;
  font-size: 14px;
  color: #4a4a4a;
  background-color: #fff;
  border: 1px solid #333333;

  &:hover {
    color: #fff;
    background-color: #2196f3;
  }

  &:active {
    color: #fff;
    background-color: #2196f3;
  }
`;
