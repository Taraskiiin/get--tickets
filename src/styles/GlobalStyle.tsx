import { createGlobalStyle } from 'styled-components';

const backgroundColor = '#f3f7fa';

export const GlobalStyle = createGlobalStyle`
  * {
    box-sizing: border-box;
    margin: 0;
    padding: 0;
  }
  body {
      background-color: ${backgroundColor};
  }
  .container{
      padding: 0 50px;
  }
  @media (max-width: 768px) {
      .container {
        padding: 0 10px;
      }
  }
  `;
