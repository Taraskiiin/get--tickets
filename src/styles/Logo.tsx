import styled from 'styled-components';

export const NavLink = styled.a`
  margin: 50px auto;
  width: 20%;
  img {
    max-width: 150px;
    height: 100%;
  }
  @media ${(props) => props.theme.device.tablet} {
    width: 35%;
  }
`;
