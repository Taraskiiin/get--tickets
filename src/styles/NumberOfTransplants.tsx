import styled from 'styled-components';
import { Form, Field } from 'formik';
import { Button } from 'react-bootstrap';

export const NumberOfTransplantsBlock = styled.div`
  height: max-content;
  display: flex;
  padding: 20px;
  flex-direction: column;
  justify-content: space-around;
  align-items: start;
  background-color: ${(props) => props.theme.colors.white};
  border-radius: ${(props) => props.theme.borderRadius};
  font-size: ${(props) => props.theme.fontSizes.m};
  margin-bottom: 20px;
  box-shadow: 3px 3px 2px 1px ${(props) => props.theme.colors.athensGray};

}
  @media ${(props) => props.theme.device.tablet} {
    font-size: ${(props) => props.theme.fontSizes.s};
    padding: 10px;
  }
  @media ${(props) => props.theme.device.mobileL} {
    font-size: ${(props) => props.theme.fontSizes.m};
    padding: 20px;
  }
`;

export const TitleNumberOfTransplantsBlock = styled.h3`
  font-size: ${(props) => props.theme.fontSizes.m};
  font-weight: normal;
  line-height: 10px;
  margin-bottom: 20px;
`;

export const FormStyled = styled(Form)`
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  > label {
    width: max-content;
  }
`;

export const CustomCheckbox = styled(Field)`
  -webkit-appearance: none;
  appearance: none;
  background-color: ${(props) => props.theme.colors.white};
  width: 14px;
  height: 14px;
  border: 0.5px solid ${(props) => props.theme.colors.dodgerBlue};
  cursor: pointer;
  margin-bottom: 5px;

  &::after {
    content: 'âœ“';
    color: ${(props) => props.theme.colors.dodgerBlue};
    margin-left: 1px;
    opacity: 0;
  }
  &:hover {
    background-color: ${(props) => props.theme.colors.catskillWhite};
  }
  &:checked:after {
    opacity: 1;
  }
`;

export const LabeltoCustomCheckbox = styled.label`
  cursor: pointer;
  color: ${(props) => props.theme.colors.codGray};
`;

export const SubmitBtn = styled(Button)`
height: max-content;
  max-width: 20px 
  text-align: center;
  border: none;
  background-color: ${(props) => props.theme.colors.whitecodGray};
  color: ${(props) => props.theme.colors.white};
  font-size: ${(props) => props.theme.fontSizes.s};
  text-transform: uppercase;
  
  @media ${(props) => props.theme.device.tablet} {
    max-width: 90%;
    padding: 2px 3px 2px 3px
  }
  @media ${(props) => props.theme.device.mobileL} {
    font-size: ${(props) => props.theme.fontSizes.m};
    padding: 5px 10px 5px 10px;
  }
`;
