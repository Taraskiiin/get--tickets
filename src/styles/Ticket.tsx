import styled from 'styled-components';

export const Card = styled.div`
  background-color: #fff;
  height: max-content;
  border-radius: 10px;
  padding: 25px 50px 25px 20px;
  text-transform: uppercase;
  font-size: 12px;
  margin-bottom: 20px;
`;

export const Header = styled.div`
  display: grid;
  grid-template-columns: 2fr 1fr;
  align-items: center;
  font-size: 12px;
  margin-bottom: 10px;
`;

export const Content = styled.section`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
`;

export const ColumnInfo = styled.p`
  margin-bottom: 20px;
  > span {
    display: block;
  }
`;

export const ColumnInfoTitle = styled.span`
  color: #a0b0b9;
`;

export const ColumnInfoTime = styled.span`
  text-transform: lowercase;
`;

export const Price = styled.span`
  font-size: 18px;
  color: #2196f3;
`;
