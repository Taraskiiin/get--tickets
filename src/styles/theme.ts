export interface ITheme {
  colors: Record<string, string>;
  fontSizes: Record<string, string>;
  borderRadius: string;
  device: Record<string, string>;
}

export const theme: ITheme = {
  colors: {
    catskillWhite: '#f3f7fa',
    white: '#ffffff',
    dodgerBlue: '#2196f3',
    gullGray: '#A0B0B9',
    codGray: '#111111',
    athensGray: '#E5E9EC',
  },
  fontSizes: {
    l: '18px',
    m: '14px',
    s: '12px',
  },
  borderRadius: '10px',
  device: {
    mobileS: '(max-width: 320px)',
    mobileM: '(max-width: 375px)',
    mobileL: '(max-width: 425px)',
    tablet: '(max-width: 768px)',
    laptop: '(max-width: 1024px)',
    laptopL: '(max-width: 1440px)',
    desktop: '(max-width: 2560px)',
  },
};
