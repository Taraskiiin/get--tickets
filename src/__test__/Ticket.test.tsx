import React from 'react';
import renderWith from './RenderWith';
import Ticket from '../components/Ticket';
import { theme } from '../styles/theme';

const propsToTicketsComponent = {
  carrier: 'EK',
  price: 20895,
  segments: [
    {
      date: '2022-01-14T17:19:00.000Z',
      destination: 'HKT',
      duration: 1433,
      origin: 'MOW',
      stops: ['DXB', 'HKG'],
    },
    {
      date: '2022-02-03T12:11:00.000Z',
      destination: 'MOW',
      duration: 639,
      origin: 'HKT',
      stops: ['HKG', 'DXB', 'AUH'],
    },
  ],
};

describe('Ticket /> testing', () => {
  it('Should render with ThemeProvider', () => {
    expect(
      renderWith.withTheme(<Ticket {...propsToTicketsComponent} />, theme)
    ).toMatchSnapshot();
  });
});
