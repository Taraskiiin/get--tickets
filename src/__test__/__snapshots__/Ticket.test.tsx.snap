// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticket /> testing Should render with ThemeProvider 1`] = `
<div
  className="sc-bdvvtL juwDzc"
>
  <div
    className="sc-gsDKAQ emeXrs"
  >
    <span
      className="sc-gKclnd dKkrHJ"
    >
      20895
       P
    </span>
    <span
      className="sc-iCfMLu cYmdZd"
    >
      EK
    </span>
  </div>
  <section
    className="sc-dkPtRN ggcBqZ"
  >
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        MOW
         - 
        HKT
      </span>
      <span>
        Jan 14th 22, 7:19
      </span>
    </p>
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        В Пути
      </span>
      <span
        className="sc-jRQBWg hgwUXy"
      >
        23Ч 53М
      </span>
    </p>
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        Пересадки
      </span>
      <span>
        DXB
         
      </span>
      <span>
        HKG
         
      </span>
    </p>
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        HKT
         - 
        MOW
      </span>
      <span>
        Feb 3rd 22, 2:11
      </span>
    </p>
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        В Пути
      </span>
      <span
        className="sc-jRQBWg hgwUXy"
      >
        10Ч 39М
      </span>
    </p>
    <p
      className="sc-hKwDye dwqymA"
    >
      <span
        className="sc-eCImPb ijgQPv"
      >
        Пересадки
      </span>
      <span>
        HKG
         
      </span>
      <span>
        DXB
         
      </span>
      <span>
        AUH
         
      </span>
    </p>
  </section>
</div>
`;
